src/desktop/Constraint.cpp

```diff
@@ -62,8 +63,12 @@ void CConstraint::onCommit() {
     const auto COMMITTED = m_pConstraint->current.committed;
 
     if (COMMITTED & WLR_POINTER_CONSTRAINT_V1_STATE_CURSOR_HINT) {
-        m_bHintSet      = true;
-        m_vPositionHint = {m_pConstraint->current.cursor_hint.x, m_pConstraint->current.cursor_hint.y};
+        static auto PXWLFORCESCALEZERO = CConfigValue<Hyprlang::INT>("xwayland:force_zero_scaling");
+
+        m_bHintSet       = true;
+        const auto ISXWL = m_pOwner->getWindow()->m_bIsX11;
+        const auto SCALE = ISXWL && *PXWLFORCESCALEZERO ? m_pOwner->m_fLastScale : 1;
```

Comment:
you should use the window property for xwayland scale, forgot the name, iirc x11scaledby

Reply:
Got it.